### ISSUE TITLE
8.2.2 Sharing search inconsistent with internal users

### ISSUE THREAD

**Issue by gig13** [owncloud/core#22920]
### Steps to reproduce
1. Setup LDAP with an AD
2. Click to Share a file/folder and start typing each letter -- checking for results

### Expected behavior
After typing in two or three letters, a consistent result set should be supplied for selecting a user/group

### Actual behavior
The search string khopkins should show two accounts for my colleague Kevin. One is a regular user account with displayname Hopkins, Kevin and the other is an admin account with displayname Kevin Hopkins. They are in separate AD domains.

Type in k - no spinner - expected.
Add h - no results
Add o - no results
Add p - 2 results - expected.
Add k - no results
Add i - 2 results
Add n - 2 results
Add s - 2 results
From back of string
Del s - no results
Del n - no results
Del i - no results
Del k - no results
Del p - 2 results
Del o - 2 results

This is the result I get when it fails:
```
{"data":[],"status":"success"}
```

This is the result I get when is actually succeeds:
````
{"data":[{"label":"Hopkins, Kevin","value":{"shareType":0,"shareWith":"13C04BB8-6C06-4D50-A90A-14DD2080D753"}},{"label":"Kevin Hopkins","value":{"shareType":0,"shareWith":"06D6E22E-1FBB-4DEA-A1C7-DA16879C89E5"}}],"status":"success"}
```

The above is not exactly reproducible. For example if I run it another time it might not work at all as I add each character and then with I remove the trailing s it will work, then no more successes. Sometimes it can take several minutes to be able to get another success, constantly adding and removing characters, even trying to search for another user.

Sometimes it doesn't work at all.


### Server configuration
**Operating system**: RHEL 7

**Web server:** Apache

**Database:**  Galera

**PHP version:** php5.5

**ownCloud version:** OC8.2.2

**Updated from an older ownCloud or fresh install:** Updated from 8.1.5

**Where did you install ownCloud from:** Repositories (Enterprise)

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** ActiveDirectory


#### ownCloud log (data/owncloud.log)
On s3
https://s3.owncloud.com/owncloud/index.php/apps/files?dir=%2FShared%2Fowncloud%2Fsupport%2Fgithub-issues%2Fcore%2F22920

@MorrisJobke

**Comment by PVince81** [comment]
@blizzz

**Comment by blizzz** [comment]
Most likely yet another dup of #21896 and fixed by #21999, to be shipped with 8.2.3.

**Comment by MorrisJobke** [comment]
> Most likely yet another dup of #21896 and fixed by #21999, to be shipped with 8.2.3.

Yes. Seems so. @gig13 Please reopen this if this still happens with 8.2.3.

**Comment by MorrisJobke** [comment]
> Click to Share a file/folder and start typing each letter -- checking for results

Easiest way to reproduce: Type in 2 letters - wait for the autocomplete list - type another letter - remove this last letter so that the initial 2 letters are in - wait for the list (before: empty, after: same result as in the beginning)
